gradle/detect.go

```diff
@@ -95,7 +96,44 @@ func (Detect) Detect(context libcnb.DetectContext) (libcnb.DetectResult, error)
 					},
 				},
 			},
-		}, nil
+		}
+	}
+	// Gradle's detection has passed
+	if len(result.Plans) > 0 {
+		if cr.ResolveBool("BP_JAVA_INSTALL_NODE") {
+			pathToSearch := context.Application.Path
+			files := []string{"yarn.lock", "package.json"}
+			var requireYarn, requireNode bool
+
+			if customNodePath, _ := cr.Resolve("BP_NODE_PROJECT_PATH"); customNodePath != "" {
+				pathToSearch = filepath.Join(context.Application.Path, customNodePath)
+			}
+			for _, file := range files {
+				_, err := os.Stat(filepath.Join(pathToSearch, file))
+				if os.IsNotExist(err) {
+					continue
+				} else if err != nil {
+					return libcnb.DetectResult{}, fmt.Errorf("unable to determine if file %s exists \n%w", file, err)
+				}
+				if file == "yarn.lock" {
+					requireYarn = true
+					break
+				}
+				if file == "package.json" {
+					requireNode = true
+				}
+			}
```

Comment:
This logic is being duplicated. You have a very similar loop above to find the gradle files.

I would suggest extracting the logic into a `findFirstFile` method that takes the list of files to search and a callback method `runWhenFound` which would get called back with the name of the file found.

You could go a step further and extract methods for `findGradleFiles` and `fileNodeFiles` that wrap up more logic.

It's a little more work, but it would improve the readability of the code.